util/unionfind.h

```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+// Copyright (c) 2018-present The Alive2 Authors.
+// Distributed under the MIT license that can be found in the LICENSE file.
+
+#include <vector>
+
+namespace util {
+
+class UnionFind final {
+  std::vector<unsigned> id;
+  unsigned root(unsigned i);
+public:
+  UnionFind(unsigned n);
+  unsigned find(unsigned i);
+  void unite(unsigned p, unsigned q);
```

Comment:
this is usually spelled union()

Reply:
Oh I was using unite to avoid the conflict with the keyword union. What should we call it, is uf_union better?

Reply:
ah, right 😵
then let's leave it like that 👍